lib/steps.js

```diff
@@ -576,6 +577,108 @@ function detectCompendium(passon) {
     });
 }
 
+function fetchCompendiumID(passon) {
+    return new Promise((fulfill, reject) => {
+        if (passon.isCompendium) {
+            debug('[%s] Reading compendium ID from %s', passon.id, config.compendium.detectionFileName);
+
+            let compendiumID = null;
+
+            // Get compendium yml, or throw exception on error
+            try {
+                let doc = yaml.safeLoad(fs.readFileSync(passon.detectionFile, 'utf8'));
+                compendiumID = doc.id;
+                debug('[%s] Successfully read compendium metadata: %O', passon.id, doc);
+
+            } catch (error) {
+                debug('[%s] Could not read compendium detection file %s: %s', passon.id, passon.detectionFile, error);
+                error.msg = 'Could not read compendium detection file';
+                error.status = 400;
+                reject(error);
+            }
+
+            if (!compendiumID){
+                debug('[%s] No ID specified for compendium', passon.id);
+                let message = 'No id found in compendium detection file';
+                let error = new Error(message);
+                error.msg = message;
+                error.status = 400;
+                reject(error);
+            } else {
+                // Validate compendium id
+                let idRegex = new RegExp(config.bagtainer.id_regex);
```

Comment:
I think `new RegExp` is not neccessary, you can directly do `config.bagtainer.id_regex.test(compendiumID)`.